.github/workflows/go.yml

```diff
@@ -188,9 +188,13 @@ jobs:
           cache: true
           cache-dependency-path: go/go.sum
       - name: Run build
-        run: |
-          cd go
-          GOARCH=386 go build ./...
+        run: GOARCH=386 go build ./...
+        working-directory: ./go
+      - name: Run test
+        # No asm simd support for 386
+        # WIP refactor, only tests in the specified dirs have been fixed
```

Comment:
does it actually fail without the `noasm` tag? it should be correctly detecting the architecture and defaulting to the pure go implementations even without the `noasm` tag. We should fix it otherwise

Reply:
It was failing because we didn't have a `386` variant of the BMI functions. I just added one and now it works without the build tag. This issue likely still exists with the other arch-specific function implementations but I think it's reasonable to go and add those together with the corresponding updates to `refCount` across other parts of the repo.

Reply:
bah, must have missed the BMI stuff when I added more the *default* impls for architectures I didn't cover -_-